src/Raven.Server/Documents/DatabaseRequestHandler.cs

```diff
@@ -18,6 +18,7 @@ public override void Init(RequestHandlerContext context)
             Database = context.Database;
             ContextPool = Database?.DocumentsStorage?.ContextPool;
             IndexStore = context.Database.IndexStore;
+            _logger = Database.LoggerSetup.GetLogger<DatabaseRequestHandler>(Database.Name);
```

Comment:
This is called per connection, let us see if we can cache that in some manner? 
Also, this is going to have ALL the handlers using the same logger, which is something we want to avoid